Workflow/Sources/WorkflowHost.swift

```diff
@@ -80,11 +78,69 @@ public final class WorkflowHost<WorkflowType: Workflow> {
         handle(output: output)
     }
 
+    /// Add a listener closure to be called back when a render pass happens.
+    public func addRenderingListener(_ closure: @escaping (WorkflowType.Rendering) -> Void) -> UUID {
+        dispatchPrecondition(condition: .onQueue(DispatchQueue.main))
```

Comment:
Just want to verify we do not want to be on `DispatchQueue.workflowExecution`?

Reply:
Dhaval said that workflow host should always be called on the main thread so that's why I used `main` instead of `workflowExecution` since that could be changed in the future to something else.  Also `workflowExecution` is used internally in Workflow to process node updates and these public functions are called from external sources.
Since currently these equate to the same thing I'm open to change it but I was assuming `workflowExecution` was more of an internal thing to workflow.